pkg/dynamicinstrumentation/ditypes/analysis.go

```diff
@@ -34,14 +34,16 @@ type TypeMap struct {
 
 // Parameter represents a function parameter as read from DWARF info
 type Parameter struct {
-	Name             string
-	ID               string
-	Type             string
-	TotalSize        int64
-	Kind             uint
-	Location         Location
-	NotCaptureReason NotCaptureReason
-	ParameterPieces  []Parameter
+	Name                string
+	ID                  string
+	Type                string
+	TotalSize           int64
+	Kind                uint
+	Location            *Location
+	LocationExpressions []LocationExpression
+	FieldOffset         uint64
+	NotCaptureReason    NotCaptureReason
```

Comment:
I would just call it `Reason`

Reply:
This is what it's called in other dynamic instrumentation implementations.